app/services/page_creator.rb

```diff
@@ -45,16 +33,22 @@ def initialize(page, params)
 
     private
 
+    def page_exists
+      !@page.send("ak_#{page_type}_resource_uri").blank?
+    end
+
     def handle_response(response)
       if !response.success?
-        @page.update!(status: 'failed', messages: response.parsed_response)
-        raise Error.new("HTTP Response code: #{response.code}, body: #{response.body}")
+        page_update = {status: 'failed', messages: response.parsed_response}
+        Rails.logger.error("HTTP Response code: #{response.code}, body: #{response.body}")
+      else
+        page_update = {
+            "ak_#{page_type}_resource_uri" => response.headers['location'],
```

Comment:
This is where the petition page / donation page uri is located in the response headers